src/librustc_trans/trans/intrinsic.rs

```diff
@@ -589,217 +582,86 @@ pub fn trans_intrinsic_call<'a, 'blk, 'tcx>(mut bcx: Block<'blk, 'tcx>,
-                                    "llvm.smul.with.overflow.i32",
-                                    llargs[0],
-                                    llargs[1],
-                                    llresult,
-                                    call_debug_location),
-        (_, "i64_mul_with_overflow") =>
-            with_overflow_intrinsic(bcx,
-                                    "llvm.smul.with.overflow.i64",
-                                    llargs[0],
-                                    llargs[1],
-                                    llresult,
-                                    call_debug_location),
-        (_, "u8_mul_with_overflow") =>
-            with_overflow_intrinsic(bcx,
-                                    "llvm.umul.with.overflow.i8",
-                                    llargs[0],
-                                    llargs[1],
-                                    llresult,
-                                    call_debug_location),
-        (_, "u16_mul_with_overflow") =>
-            with_overflow_intrinsic(bcx,
-                                    "llvm.umul.with.overflow.i16",
-                                    llargs[0],
-                                    llargs[1],
-                                    llresult,
-                                    call_debug_location),
-        (_, "u32_mul_with_overflow") =>
-            with_overflow_intrinsic(bcx,
-                                    "llvm.umul.with.overflow.i32",
-                                    llargs[0],
-                                    llargs[1],
-                                    llresult,
-                                    call_debug_location),
-        (_, "u64_mul_with_overflow") =>
-            with_overflow_intrinsic(bcx,
-                                    "llvm.umul.with.overflow.i64",
-                                    llargs[0],
-                                    llargs[1],
-                                    llresult,
-                                    call_debug_location),
-
-        (_, "unchecked_udiv") => UDiv(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "unchecked_sdiv") => SDiv(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "unchecked_urem") => URem(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "unchecked_srem") => SRem(bcx, llargs[0], llargs[1], call_debug_location),
-
-        (_, "overflowing_add") => Add(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "overflowing_sub") => Sub(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "overflowing_mul") => Mul(bcx, llargs[0], llargs[1], call_debug_location),
+        (_, c @ "ctlz") | (_, c @ "cttz") | (_, c @ "ctpop") | (_, c @ "bswap") |
+        (_, c @ "add_with_overflow") | (_, c @ "sub_with_overflow") | (_, c @ "mul_with_overflow") |
+        (_, c @ "overflowing_add") | (_, c @ "overflowing_sub") | (_, c @ "overflowing_mul") |
+        (_, c @ "unchecked_div") | (_, c @ "unchecked_rem") => {
```

Comment:
You don't need all the `c @ ...` because `c` is already in the `name` variabile.